src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -164,6 +167,77 @@ class spell_pri_aq_3p_bonus : public SpellScriptLoader
         }
 };
 
+// Last Update 7.2.0
+// Called by  Plea - 200829, Power Word: Shield - 17, Shadow Mend - 186263, Power Word: Radiance  - 194509
+// Atonement - 81749
+class spell_pri_atonement : public SpellScriptLoader
+{
+public:
+    spell_pri_atonement() : SpellScriptLoader("spell_pri_atonement") { }
+
+    class spell_pri_atonement_AuraScript : public AuraScript
+    {
+        PrepareAuraScript(spell_pri_atonement_AuraScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            return ValidateSpellInfo({ 
+                SPELL_PRIEST_ATONEMENT_AURA_TRIGGERED,
+                SPELL_PRIEST_ATONEMENT_HEAL
+            });
+        }
+
+        bool CheckProc(ProcEventInfo& eventInfo)
+        {
+            return GetCaster() && eventInfo.GetDamageInfo();
+        }
+
+        void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+        {
+            Player* player = GetCaster()->ToPlayer();
+
+            SpellInfo const* spellInfoAtonement = sSpellMgr->GetSpellInfo(SPELL_PRIEST_ATONEMENT_AURA);
+            if (!spellInfoAtonement || !player->HasAura(SPELL_PRIEST_ATONEMENT_AURA))
+                return;
+
+            if (player->GetUInt32Value(PLAYER_FIELD_CURRENT_SPEC_ID) != TALENT_SPEC_PRIEST_DISCIPLINE)
```

Comment:
Don't check this - only disc priests should have this in SpecializationSpells.db2